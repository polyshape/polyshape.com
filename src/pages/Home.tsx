import { useNavigate } from 'react-router-dom';
import { AppRoutes } from '../lib/common/AppRoutes';

export default function Home() {
  const navigate = useNavigate();

  const goToVision = () => navigate(AppRoutes.VISION.path);
  const goToContact = () => navigate(AppRoutes.CONTACT.path);

  return (
    <div className="hero">
      <picture className="hero__bg">
        <source srcSet="/home_hero_transparent.webp" type="image/webp" />
        <img src="/home_hero_transparent.png" alt="Background" />
      </picture>
      <div className="hero__overlay" />
      <div className="hero__content">
        <h1 className="hero__title">Compositional multi-agent learning systems that advance <span className="highlight">health intelligence</span></h1>
        <p className="hero__subtitle">PolyShape builds modular systems on mathematical foundations, integrating uncertainty and diverse biomedical data to support monitoring, decision-making, and intervention in healthcare.</p>
      <div className="hero__buttons">
        <button className="button__primary" onClick={goToVision} aria-label="Learn more about our vision">Learn more</button>
        <button className="button__primary" onClick={goToContact} aria-label="Contact us">Get in touch</button>
      </div>
      </div>
    </div>
  );
}
