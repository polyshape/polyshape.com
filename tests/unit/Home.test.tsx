import '@testing-library/jest-dom';
import Home from '../../src/pages/Home';
import { renderWithRouterAndLoadingProviders } from './utils/renderWithProviders';

describe('Home page', () => {
  it('renders the main title and subtitle', () => {
    const { getByText } = renderWithRouterAndLoadingProviders(<Home />);
    expect(getByText(/Compositional multi-agent learning systems that advance/i)).toBeInTheDocument();
    expect(getByText(/PolyShape builds modular systems on mathematical foundations/i)).toBeInTheDocument();
  });

  it('applies correct CSS classes', () => {
    const { container } = renderWithRouterAndLoadingProviders(<Home />);
    expect(container.querySelector('.hero')).toBeInTheDocument();
    expect(container.querySelector('.hero__content')).toBeInTheDocument();
    expect(container.querySelector('.hero__title')).toBeInTheDocument();
    expect(container.querySelector('.hero__subtitle')).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { container } = renderWithRouterAndLoadingProviders(<Home />);
    expect(container).toMatchSnapshot();
  });
});
