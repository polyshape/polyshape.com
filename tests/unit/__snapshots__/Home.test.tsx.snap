// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Home page matches the snapshot 1`] = `
<div>
  <div
    class="hero"
  >
    <div
      class="hero__content"
    >
      <h1
        class="hero__title"
      >
        Generalized Health Intelligence
      </h1>
      <p
        class="hero__subtitle"
      >
        Building the Intelligence that Safeguards Us!
      </p>
    </div>
  </div>
</div>
`;
